src/sentry/api/endpoints/organization_events.py

```diff
@@ -35,6 +35,23 @@
 
 
 class OrganizationEventsV2Endpoint(OrganizationEventsV2EndpointBase):
+    def has_feature_for_fields(self, feature, organization, request, fields):
```

Comment:
the definition for this function is a bit confusing to me, could we change it perhaps to something like and add a docstring too?
```suggestion
    def has_feature_for_fields(self, feature, organization, request, feature_fields):
```

Also is this correct at the endpoint level? `request.GET.get("query")` isn't parsed yet so iterating on it will result in iterating character by character, I think this might fit better during query parsing in `get_filter` and field parsing in `resolve_field_list`

Reply:
I just figured we'd raise this error as close to the request as possible- this is just checking if any of the fields are in the query string- something like
`any(field in 'event.type:transaction user_misery_new():>0.050' for field in feature_fields)`

Reply:
I think we still have to at least perform the query parsing because of something like this:
```python
any(field in 'tags[something]:project_threshold_config' for field in feature_fields)
or
any(field in 'message:"user_misery_new()"' for field in feature_fields)
```

Reply:
hmm yeah you're right, I'll update

Reply:
Gah ok, I can't find a nice way to do it. I'll need to query either query postgres to get the organization based off the organization_id in the params to check the feature flag (meh) or change the function signatures to pass organization down to `get_filter` or somewhere where it makes sense to check this which I don't want to do. I might just remove the query check and leave a todo for now?

Reply:
:+1: I think this would be great as a separate PR given the complexity involved